pkg/csi/service/node.go

```diff
@@ -338,11 +338,28 @@ func (driver *vsphereCSIDriver) NodeGetInfo(
 
 	var nodeInfoResponse *csi.NodeGetInfoResponse
 
+	var nodeID string
+	var err error
 	nodeName := os.Getenv("NODE_NAME")
 	if nodeName == "" {
-		return nil, logger.LogNewErrorCode(log, codes.Internal, "ENV NODE_NAME is not set")
+		return nil, logger.LogNewErrorCode(log, codes.Internal,
+			"ENV NODE_NAME is not set")
+	}
+	if commonco.ContainerOrchestratorUtility.IsFSSEnabled(ctx, common.UseCSINodeId) {
+		// Get VM UUID
+		nodeID, err = driver.osUtils.GetSystemUUID(ctx)
+		if err != nil {
+			return nil, logger.LogNewErrorCodef(log, codes.Internal,
+				"failed to get system uuid for node VM with error: %v", err)
+		}
+		nodeID, err = driver.osUtils.ConvertUUID(nodeID)
```

Comment:
@BaluDontu I believe this line is causing the issue described in https://github.com/kubernetes-sigs/vsphere-csi-driver/issues/1629

Could you have a look?